### PR TITLE
Add file handle leak detection

### PR DESCRIPTION
**Pull request by Wadeck**
Based on a collaborative effort between Jeff Thompson and me.

This non-regression test should prevent us to encounter again issues like [JENKINS-64632](https://issues.jenkins.io/browse/JENKINS-64632) or [JENKINS-61121](https://issues.jenkins.io/browse/JENKINS-61121).
It was tested with the correct (current) logic and with the one before the correction to ensure the test can uncover such failure before the end user.

### Proposed changelog entries

* No change log

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] (If applicable) Jira issue is well described
- [n/a] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java
Comment (timja) [inline test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java:30]: This type is exported and is safe to use 

https://docs.oracle.com/javase/8/docs/jre/api/management/extension/com/sun/management/UnixOperatingSystemMXBean.html

https://docs.oracle.com/javase/8/docs/jdk/api/javac/tree/jdk/Exported.html

**Review by timja — COMMENTED** [review]
Looks good if I have time will look through more later

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
